src/fluree/db/ledger.cljc

```diff
@@ -9,70 +9,51 @@
             [fluree.db.nameservice :as nameservice]
             [fluree.db.util :as util :refer [get-first get-first-value]]
             [fluree.db.util.async :refer [<? go-try]]
+            [fluree.db.util.ledger :as util.ledger]
             [fluree.db.util.log :as log]))
 
 #?(:clj (set! *warn-on-reflection* true))
 
-(defn get-branch-meta
-  "Retrieves branch metadata from ledger state"
-  [{:keys [state] :as _ledger} requested-branch]
-  (let [{:keys [branch branches]} @state]
-    (if requested-branch
-      (get branches requested-branch)
-      ;; default branch
-      (get branches branch))))
-
-(defn available-branches
-  [{:keys [state] :as _ledger}]
-  (-> @state :branches keys))
-
 ;; TODO - no time travel, only latest db on a branch thus far
 (defn current-db
-  ([ledger]
-   (current-db ledger nil))
-  ([ledger branch]
-   (if-let [branch-meta (get-branch-meta ledger branch)] ; if branch is nil, will return default
-     (branch/current-db branch-meta)
-     (throw (ex-info (str "Invalid branch: " branch " is not one of:"
-                          (available-branches ledger))
-                     {:status 400, :error :db/invalid-branch})))))
+  "Returns the current database for this ledger.
+  Since each ledger now represents a single branch, no branch parameter is needed."
+  [ledger]
+  (when-let [state (:state ledger)]
```

Comment:
`:state` is always present on the Ledger record.

Reply:
Addressed in a161d37baca16ba5caefc3e0f6234f1b39351949